src/rpc/blockchain.cpp

```diff
@@ -180,7 +179,7 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* tip, const CBlockIn
         case TxVerbosity::SHOW_DETAILS:
         case TxVerbosity::SHOW_DETAILS_AND_PREVOUT:
             CBlockUndo blockUndo;
-            const bool have_undo{WITH_LOCK(::cs_main, return !IsBlockPruned(blockindex) && UndoReadFromDisk(blockUndo, blockindex))};
+            const bool have_undo{WITH_LOCK(::cs_main, return !blockman.IsBlockPruned(blockindex)) && UndoReadFromDisk(blockUndo, blockindex)};
```

Comment:
In commit "move-mostly: Make fHavePruned a BlockMan member" (7123f3d8af4e592212439968f4eaaec912ec8d02)

Possible bug or unrelated change? Before this locked cs_main while calling UndoReadFromDisk, now it now longer does. It does seems good to not lock cs_main while calling UndoReadFromDisk, but it doesn't really make sense as part of this commit.

Reply:
Ah! Yes this was unintentional 😅 
Fixed in 85b8b25239f6a7ce1f6877eec52d786352c90872